File: AspNetMvcUnleashed/0672329980_source/Chapter21/VB/UnleashedBlog/Controllers/ArchiveController.vb

Public Class ArchiveController
    Inherits ApplicationController
    Public Sub New()
    End Sub

    Public Sub New(ByVal blogRepository As BlogRepositoryBase)
        MyBase.New(blogRepository)
    End Sub


    ''' <summary>
    ''' Returns a single blog entry
    ''' </summary>
    Public Function Details(ByVal year As Integer, ByVal month As Integer, ByVal day As Integer, ByVal name As String) As ActionResult
        Return View(_blogService.GetBlogEntry(year, month, day, name))
    End Function

    ''' <summary>
    ''' Returns a list of blog entries that match a specified year, month, or day
    ''' </summary>
    Public Function Index(ByVal page As Nullable(Of Integer), ByVal year As Nullable(Of Integer), ByVal month As Nullable(Of Integer), ByVal day As Nullable(Of Integer)) As ActionResult
        Return View(_blogService.ListBlogEntries(page, year, month, day))
    End Function

    ''' <summary>
    ''' Only callable within an Ajax request.
    ''' Returns a list of blog entries that match a specified year, month, or day.
    ''' </summary>
    <AcceptAjax(), ActionName("Index")> _
    Public Function Index_Ajax(ByVal page As Nullable(Of Integer), ByVal year As Nullable(Of Integer), ByVal month As Nullable(Of Integer), ByVal day As Nullable(Of Integer)) As ActionResult
        Return PartialView("BlogEntries", _blogService.ListBlogEntries(page, year, month, day))
    End Function


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter21/CS/UnleashedBlog/UnleashedBlog/Controllers/ArchiveController.cs
using System.Web.Mvc;
using UnleashedBlog.Models;
using UnleashedBlog.Models.EntityFramework;
using UnleashedBlog.Attributes;

namespace UnleashedBlog.Controllers
{
    public class ArchiveController : ApplicationController
    {
        public ArchiveController()
        { }

        public ArchiveController(BlogRepositoryBase blogRepository) 
            : base(blogRepository) { }


        /// <summary>
        /// Returns a single blog entry
        /// </summary>
        public ActionResult Details(int year, int month, int day, string name)
        {
            return View(_blogService.GetBlogEntry(year, month, day, name));
        }

        /// <summary>
        /// Returns a list of blog entries that match a specified year, month, or day
        /// </summary>
        public ActionResult Index(int? page, int? year, int? month, int? day)
        {
            return View(_blogService.ListBlogEntries(page, year, month, day));
        }

        /// <summary>
        /// Only callable within an Ajax request.
        /// Returns a list of blog entries that match a specified year, month, or day.
        /// </summary>
        [AcceptAjax]
        [ActionName("Index")]
        public ActionResult Index_Ajax(int? page, int? year, int? month, int? day)
        {
            return PartialView("BlogEntries", _blogService.ListBlogEntries(page, year, month, day));
        }


    }
}
